api/types/device.go

```diff
@@ -254,3 +277,33 @@ func ResourceEnrollStatusFromString(enrollStatus string) (devicepb.DeviceEnrollS
 		return devicepb.DeviceEnrollStatus_DEVICE_ENROLL_STATUS_UNSPECIFIED, trace.BadParameter("unknown enroll status %q", enrollStatus)
 	}
 }
+
+func ResourceAttestationTypeToString(
+	attestationType devicepb.AttestationType,
+) string {
+	switch attestationType {
+	case devicepb.AttestationType_ATTESTATION_TYPE_UNSPECIFIED:
+		return "unspecified"
+	case devicepb.AttestationType_ATTESTATION_TYPE_HW_KEY:
+		return "hw_key"
+	case devicepb.AttestationType_ATTESTATION_TYPE_HW_CERT:
+		return "hw_cert"
+	default:
+		return attestationType.String()
+	}
+}
+
+func ResourceAttestationTypeFromString(
+	attestationType string,
+) (devicepb.AttestationType, error) {
+	switch attestationType {
+	case "unspecified":
```

Comment:
```suggestion
	case "unspecified", "":
```

so the "default" value isn't invalid. We've had similar problems with enroll status.

Reply:
yup - nice idea.

Reply:
[8280414](https://github.com/gravitational/teleport/pull/24518/commits/8280414c55ce02cb462f9b39a9f458a4b3b46937)